node/follower_node.go

```diff
@@ -251,18 +251,26 @@ func (node *AlgorandFollowerNode) GetPendingTransaction(_ transactions.Txid) (re
 }
 
 // Status returns a StatusReport structure reporting our status as Active and with our ledger's LastRound
-func (node *AlgorandFollowerNode) Status() (s StatusReport, err error) {
+func (node *AlgorandFollowerNode) Status() (StatusReport, error) {
 	node.syncStatusMu.Lock()
-	s.LastRoundTimestamp = node.lastRoundTimestamp
-	s.HasSyncedSinceStartup = node.hasSyncedSinceStartup
+	lastRoundTimestamp := node.lastRoundTimestamp
+	hasSyncedSinceStartup := node.hasSyncedSinceStartup
 	node.syncStatusMu.Unlock()
 
 	node.mu.Lock()
 	defer node.mu.Unlock()
+	var s StatusReport
+	var err error
 	if node.catchpointCatchupService != nil {
-		return catchpointCatchupStatus(node.catchpointCatchupService.GetLatestBlockHeader(), node.catchpointCatchupService.GetStatistics()), nil
+		s = catchpointCatchupStatus(node.catchpointCatchupService.GetLatestBlockHeader(), node.catchpointCatchupService.GetStatistics())
+	} else {
+		s, err = latestBlockStatus(node.ledger, node.catchupService)
 	}
-	return latestBlockStatus(node.ledger, node.catchupService)
+
+	s.LastRoundTimestamp = lastRoundTimestamp
+	s.HasSyncedSinceStartup = hasSyncedSinceStartup
```

Comment:
```suggestion
	s.LastRoundTimestamp = node.lastRoundTimestamp
	s.HasSyncedSinceStartup = node.hasSyncedSinceStartup
```

Reply:
Is there a reason to store these in the beginning and then assign them at the end?

Reply:
Just the mutexes. The status mutex unlock isn't deferred and I was hesitant to change `node.mu` to be non-deferred instead since it does more.

Reply:
Oh i see, makes sense.